services/worker/src/worker/job_runners/_datasets_based_job_runner.py

```diff
@@ -44,18 +44,18 @@ def __init__(
         self.datasets_based_config = app_config.datasets_based
         self.base_datasets_cache = hf_datasets_cache
 
-    def get_cache_subdirectory(self, date: datetime) -> str:
-        date_str = date.strftime("%Y-%m-%d-%H-%M-%S")
+    def get_cache_subdirectory(self, digits: int = 14) -> str:
+        random_str = f"{random.randrange(10**(digits - 1), 10**digits)}"  # nosec B311
```

Comment:
I would prefer to keep the date, because it allows to manually clean the old stale directories. Normally the directory is removed automatically, but if the worker crashes (which still occurs a lot) it's kept on the disk.
See #1122.

btw: I'm currently doing such a maintenance operation, removing the dangling cache directories for may:

```bash
find . -maxdepth 1 -name "2023-05*" -type d -exec rm -rf {} \;
```

With a random string only, we lose this possibility

Reply:
> removing the dangling cache directories for may

it finished, and made the inodes usage fall from 82% to 65%. So, definitively not a detail

Reply:
Removing my comments!

Reply:
The command to clean after 3 days, for example:

```bash
find . -maxdepth 1 -mtime +3 -type d -exec rm -r {} +
```